cpp/benchmarks/io/nvbench_helpers.hpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <benchmarks/common/generate_input.hpp>
+#include <benchmarks/io/cuio_common.hpp>
+
+#include <cudf/io/types.hpp>
+#include <cudf/types.hpp>
+
+#include <nvbench/nvbench.cuh>
+
+enum class data_type : int32_t {
+  INTEGRAL        = static_cast<int32_t>(type_group_id::INTEGRAL),
+  INTEGRAL_SIGNED = static_cast<int32_t>(type_group_id::INTEGRAL_SIGNED),
+  FLOAT           = static_cast<int32_t>(type_group_id::FLOATING_POINT),
+  DECIMAL         = static_cast<int32_t>(type_group_id::FIXED_POINT),
+  TIMESTAMP       = static_cast<int32_t>(type_group_id::TIMESTAMP),
+  STRING          = static_cast<int32_t>(cudf::type_id::STRING),
+  LIST            = static_cast<int32_t>(cudf::type_id::LIST),
+  STRUCT          = static_cast<int32_t>(cudf::type_id::STRUCT)
+};
```

Comment:
Why don't use enums from `cudf::type_id` directly?

Reply:
Can the types like `INTEGRAL` be just `type_id::INT32`?

Reply:
INTEGRAL represents all integral types. Types are used in round-robin order so we get a mix of columns with different, but related, types. This allows us to cover all supported types without having separate benchmarks for each type.

Reply:
> Why don't use enums from cudf::type_id directly?

As you might notice, `data_type` is a combination of `type_group_id` and `cudf::type_id` thus the wrapper layer is necessary.